cmd/webhook/main.go

```diff
@@ -143,19 +144,25 @@ func main() {
 		NodeFlags:            nodeFlags,
 		Namespace:            opt.namespace,
 		VersionConstraints:   constraint,
+
+		// we could change this to get the CertDir from the configured CertName
+		// and KeyName, but doing so does not bring us any benefits but would
+		// technically break compatibility.
+		CertDir:  "/",
+		CertName: opt.admissionTLSCertPath,
+		KeyName:  opt.admissionTLSKeyPath,
 	}.Build()
 	if err != nil {
 		log.Fatalw("Failed to create admission hook", zap.Error(err))
 	}
 
 	log.Infow("Listening", "address", opt.admissionListenAddress)
 
-	if err := srv.ListenAndServeTLS(opt.admissionTLSCertPath, opt.admissionTLSKeyPath); err != nil {
+	serverContext, cancelServerContext := context.WithCancel(context.Background())
+	if err := srv.Start(serverContext); err != nil {
 		log.Fatalw("Failed to start server", zap.Error(err))
 	}
 	defer func() {
-		if err := srv.Close(); err != nil {
-			log.Fatalw("Failed to shutdown server", zap.Error(err))
-		}
+		cancelServerContext()
 	}()
```

Comment:
I'm not sure if this is actually still needed, together with the cancel-context above

Reply:
I think you can just do

```go
serverContext, cancelServerContext := context.WithCancel(context.Background())
defer cancelServerContext()
if err := srv.Start(serverContext); err != nil {
    log.Fatalw("Failed to start server", zap.Error(err))
}
```

Reply:
not done, according to https://github.com/kubermatic/machine-controller/pull/1751#discussion_r1439205717